### PR TITLE
x509_certificate_info: move main code to module_utils to allow easier implementation of diff mode

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Subset of #150.

##### ISSUE TYPE
- Refactoring Pull Request

##### COMPONENT NAME
x509_certificate_info

### CODE REVIEW

**Comment by felixfontein** [comment]
recheck

File: plugins/module_utils/crypto/module_backends/certificate_info.py [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+        except cryptography.x509.ExtensionNotFound:
+            return None, False
+
+    def _get_ocsp_must_staple(self):
+        try:
+            try:
+                # This only works with cryptography >= 2.1
+                tlsfeature_ext = self.cert.extensions.get_extension_for_class(x509.TLSFeature)
+                value = cryptography.x509.TLSFeatureType.status_request in tlsfeature_ext.value
+            except AttributeError as dummy:
```
Comment (Ajpantuso) [inline plugins/module_utils/crypto/module_backends/certificate_info.py:310]:
```suggestion
            except AttributeError:
```

File: plugins/module_utils/crypto/module_backends/certificate_info.py [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+        module.deprecate('The module is using the PyOpenSSL backend. This backend has been deprecated',
+                         version='2.0.0', collection_name='community.crypto')
+        return backend, CertificateInfoRetrievalPyOpenSSL(module, content)
+    elif backend == 'cryptography':
+        if not CRYPTOGRAPHY_FOUND:
+            module.fail_json(msg=missing_required_lib('cryptography >= {0}'.format(MINIMAL_CRYPTOGRAPHY_VERSION)),
+                             exception=CRYPTOGRAPHY_IMP_ERR)
+        return backend, CertificateInfoRetrievalCryptography(module, content)
+    else:
+        raise Exception('Unsupported value for backend: {0}'.format(backend))
```
Comment (Ajpantuso) [inline plugins/module_utils/crypto/module_backends/certificate_info.py:544]:
```suggestion
        raise ValueError('Unsupported value for backend: {0}'.format(backend))
```

File: plugins/modules/x509_certificate_info.py [hunk 3]
```diff
@@ -848,53 +341,37 @@ def main():
...
-                    name=base_dir,
-                    msg='The directory %s does not exist or the file is not a directory' % base_dir
-                )
+    if module.params['content'] is not None:
+        data = module.params['content'].encode('utf-8')
+    else:
+        try:
+            with open(module.params['path'], 'rb') as f:
+                data = f.read()
+        except Exception as e:
```
Comment (Ajpantuso) [inline plugins/modules/x509_certificate_info.py:350]:
```suggestion
        except (IOError, OSError) as e:
```

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
Looks good with same explicit exception suggestions as the previous reviews.

**Review by Ajpantuso — APPROVED** [review]
LGTM, thanks!

**Comment by felixfontein** [comment]
@Ajpantuso thanks a lot for reviewing this! :)

**Pull request merged**